### PR TITLE
LIKA-397 refacto list_errors to be fetched from db instead of from xroad-catalog

### PR DESCRIPTION
**Pull request by bertvi**
LIKA-397 refacto list_errors to be fetched from db instead of from xroad-catalog, remove fetch of list_errors from xroad-catalog

### CODE REVIEW

File: ckanext/xroad_integration/cli.py [hunk 1]
```diff
@@ -74,16 +74,6 @@ def fetch_distinct_service_stats(ctx, days):
...
-@xroad.command()
-@click.pass_context
-@click.option(u'--days', type=int)
-def fetch_list_errors(ctx, days):
-    'Fetches X-Road errors from catalog lister'
-    flask_app = ctx.meta["flask_app"]
-    with flask_app.test_request_context():
-        utils.fetch_list_errors(days)
-
-
```
Comment (Zharktas) [inline ckanext/xroad_integration/cli.py:86]: If you remove CLI commands that fetch the errors from xroad catalog, the errors are not fetched at all.

File: ckanext/xroad_integration/logic/action.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 DEFAULT_TIMEOUT = 3  # seconds
 DEFAULT_DAYS_TO_FETCH = 1
-DEFAULT_LIST_ERRORS_HISTORY_IN_DAYS = 60
+DEFAULT_LIST_ERRORS_HISTORY_IN_DAYS = 90
```
Comment (Zharktas) [inline ckanext/xroad_integration/logic/action.py:30]: How does history in days affect anything if the errors are fetched elsewhere ?

File: ckanext/xroad_integration/logic/action.py [hunk 3]
```diff
@@ -818,45 +819,6 @@ def fetch_distinct_service_stats(context, data_dict):
...
-            return []
-
-        return error_data
-
-    except ConnectionError as e:
-        log.warn("Calling listErrors failed!")
-        log.info(e)
-        return {"success": False, "message": "Fetching errors failed."}
-
-
```
Comment (Zharktas) [inline ckanext/xroad_integration/logic/action.py:859]: How the errors are now fetched from xroad catalog ?

File: ckanext/xroad_integration/utils.py [hunk 4]
```diff
@@ -93,30 +93,6 @@ def fetch_distinct_service_stats(days):
...
-                                                                    'params': json.dumps(data_dict),
-                                                                    'message': results.get('message')})
-
-    if success:
-        print(results['message'])
-
-    else:
-        print(results['message'])
-
-
```
Comment (Zharktas) [inline ckanext/xroad_integration/utils.py:119]: This is related to CLI commands and should not be removed.

File: ckanext/xroad_integration/cli.py (hunk 1 above)
Comment (bertvi) [reply ckanext/xroad_integration/cli.py:86]: This was for listErrors, the previous error fetching is still done in "def fetch_errors(ctx)"

File: ckanext/xroad_integration/logic/action.py (hunk 3 above)
Comment (bertvi) [reply ckanext/xroad_integration/logic/action.py:859]: This was for the listErrors endpoint, the previous error fetching is still done in "def fetch_xroad_errors(context, data_dict)"

File: ckanext/xroad_integration/utils.py (hunk 4 above)
Comment (bertvi) [reply ckanext/xroad_integration/utils.py:119]: This was for the listErrors endpoint, the previous error fetching is still done in "def fetch_errors()"

File: ckanext/xroad_integration/logic/action.py (hunk 2 above)
Comment (bertvi) [reply ckanext/xroad_integration/logic/action.py:30]: This is used in "xroad_error_list(context, data_dict)" for displaying last 90 days of error logs from the db under the organization view in "X-Road errors" tab, together with pagination of 20 results per page.

**Comment by bertvi** [comment]
> The errors seems to be fetched from database now, but they don't end up in database as the whole integration was removed.

They should be saved on line 466 in action.py: "XRoadError.create(**mapped_error)"

**Comment by Zharktas** [comment]
> > The errors seems to be fetched from database now, but they don't end up in database as the whole integration was removed.
> 
> They should be saved on line 466 in action.py: "XRoadError.create(**mapped_error)"

So the new rest API is not used at all anymore ? Then why it was implemented at all in the first place ?

**Comment by bertvi** [comment]
> > > The errors seems to be fetched from database now, but they don't end up in database as the whole integration was removed.
> > 
> > 
> > They should be saved on line 466 in action.py: "XRoadError.create(**mapped_error)"
> 
> So the new rest API is not used at all anymore ? Then why it was implemented at all in the first place ?

Correct, it is not used at all. Implemented probably because of miscommunication, as it was understood and also communicated, that another endpoint is needed, which would return a more specified and smaller amount of errors, e.g. per organization and with pagination, as the amount of errors returned by the previous endpoint was too large.

**Comment by Zharktas** [comment]
> > > > The errors seems to be fetched from database now, but they don't end up in database as the whole integration was removed.
> > > 
> > > 
> > > They should be saved on line 466 in action.py: "XRoadError.create(**mapped_error)"
> > 
> > 
> > So the new rest API is not used at all anymore ? Then why it was implemented at all in the first place ?
> 
> Correct, it is not used at all. Implemented probably because of miscommunication, as it was understood and also communicated, that another endpoint is needed, which would return a more specified and smaller amount of errors, e.g. per organization and with pagination, as the amount of errors returned by the previous endpoint was too large.

So we could just revert the original PR ?

**Comment by bertvi** [comment]
> > > > > The errors seems to be fetched from database now, but they don't end up in database as the whole integration was removed.
> > > > 
> > > > 
> > > > They should be saved on line 466 in action.py: "XRoadError.create(**mapped_error)"
> > > 
> > > 
> > > So the new rest API is not used at all anymore ? Then why it was implemented at all in the first place ?
> > 
> > 
> > Correct, it is not used at all. Implemented probably because of miscommunication, as it was understood and also communicated, that another endpoint is needed, which would return a more specified and smaller amount of errors, e.g. per organization and with pagination, as the amount of errors returned by the previous endpoint was too large.
> 
> So we could just revert the original PR ?

In the previous implementation, under X-Road Errors for an organization, we had just "Other errors" and "Rest errors" and there was no pagination, just links for previous and next date. The current implementation shows all errors in pages with 20 errors per page. In addition, the listErrors on the X-Road Catalog side was implemented to be able to actually view the errors, because  using GetErrors resulted in just too many errors. So depends  on, what the customer needs to be done with the errors exactly, if they are fine with how things were before, then I have just wasted time with this implementation and it can be reverted.

**Review by Zharktas — APPROVED** [review]

**Pull request merged**